### DISCUSSION TITLE
I got it working just fine on Proxmox container - Thanks for a great build!

### DISCUSSION THREAD

**Discussion by DeathGhosts** [PR3SIDENT/enshrouded-server#34] [General]
first of all let me break it down into pieces as many seams to be using Proxmox, just as me so for those who are still struggeling with making the server work, i'll list down my successful approach.

On the proxmox i went into shell to download a template  `ubuntu-23.10-standard_23.10-1_amd64.tar.zst`
to do so you'll need these lines:

Update Template list
`pveam update`

List avaliable templates
`pveam available`

Download the build you need, replace` local `with your template storage area, tab works for auto-complete if you are uncertain of its spelling:

`pveam download local ubuntu-23.10-standard_23.10-1_amd64.tar.zst`

----------------------------------------------------------------------------------------------------------------------------------
Alright, i take it you have configured your container, and one thing in mind... for mine, `i removed tick in "unprivileged container"` upon creation.

### For unprivileged container, follow current steps before installing!

`en_US.UTF-8` needs to be present on the unpriv. container in order for `steamcmd` to use `force_install_dir` correctly.

- use following command:
`nano /etc/locale.gen`

find `# en_US.UTF-8` and remove the hashtag and leading space to only `en_US.UTF-8`

Save file using `CTRL+S` followed by exiting text editor with `CTRL-X`

use command `locale-gen` to update changes!

it will appear something like this if correctly done:

root@Gameserver:# locale-gen
Generating locales (this might take a while)...
  en_US.UTF-8... done
Generation complete.
root@GameServer:# 

#### If you did install this on an unpriv. container, there is a high risk that steamcmd have installed enshrouded in `/root/.local/share/Steam/steamcmd`

To remove it,  use command: `rm -r /root/.local/share/Steam/steamcmd/enshrouded/`

————————
### Ready to continue with install / updating system

from there its very important before you dive into the whole install that you do a proper update. a golden rule that i often tend to forget:
`apt-get update`
 
from there its all about getting git installed with the command:
`apt-get install git`

once completed you can start moving into where i cloned my files to.

`cd //home && git clone https://github.com/PR3SIDENT/enshrouded-server.git`

After that you also need to set your rights to use the installer as you don't have excecutable rights to the installer. 
this line will take care of the rights and start the installer. the installer does take a while, so i created a snapshot after it completed before i went further.

`cd //home/enshrouded-server && chmod 777 InstallServer.sh  README.md && ./InstallServer.sh`

from here you will be promted with servername, password and number of server limit 0-16

to start the server you'll use this line:

`cd //home/steam/enshrouded && ./StartEnshroudedServer.sh `

or

`cd //home/steam/enshrouded && wine64 ./enshrouded_server.exe`

once it say `[Session] 'HostOnline' (up)!` then you should be a happy camper!! only port forwarding is required and you can start enjoying the fine game..

to update the game, you'll need to navigate to the steamcmd where you add the normal launch command for apt_update etc.. OR
for unpriv container, i use this command that i built inside a simple "update.sh" file:

```
#!/bin/sh
/usr/games/steamcmd +@sSteamCmdForcePlatformType windows +force_install_dir /home/steam/enshrouded +login anonymous +app_update 2278520 +quit
```

i hope this breakdown works for you also as it did for me, i am very happy that i dont need a full VM just for hosting this game.. 
upon launch, it used 820Mb of ram vs 28Gb on a full VM and the container used 4 cores at around 30% load while the full VM had 12 cores with 50-60% load.

thanks for reading and again.. i hope this breakdown works for you also!!

//DeathGhosts

**Comment by maloycolbert** [comment]
I did this and it worked out great, thanks!

However, my server didnt show up on the list even after waiting the full time for the list to load and waiting over 10 minutes to ensure the server was properly finished booting. I realized that enshrouded had a small update recently, attempted to update the app by running the steamcmd and then app_update 2278520 validate and that broke the server.

Do I need to update the server files a different way due to it being on linux?

Im currently removing all the files and steam user and reinstalling and setting my IP address in the json to see if that mightve been the issue, so ill report back if it was not an un-updated enshrouded server file

EDIT: and on uninstalling and reinstalling, im now getting an InitGameServer error, so i guess it broke even more lol.

I will note, i didnt go through your first few steps as i dont have a proxmox server, i have my own dedicated machine in my living room running ubuntu

**Comment by DeathGhosts** [comment]
The ip address in the .json config file has to be 127.0.0.1  and steamCMD will sort the rest

**Reply by maloycolbert** [reply]
even if someones internal address isnt set at 127?

for instance, my internal is 192.168.x.x

**Reply by DeathGhosts** [reply]
My server has 127.0.0.1 and during init of the server, it pulls my external ip address. 92.x.x.x

**Comment by linkandzelda91** [comment]
Hey, which port(s) require forwarding?

**Reply by DeathGhosts** [reply]
Per default its 15636 and 15637 but these can be changed if you are planning to run multiple servers

**Comment by PlutoniumJones** [comment]
Signed in just to say thank you! I was having issues with I'm guessing the container being unprivileged. I remade the container following your instructions and it worked first try.

Also when port forwarding make sure you forward TCP and UDP!

**Comment by DeathGhosts** [comment]
For those following this thread, i have troubleshooted an issue with unprivileged containers and updated this guide with the solution to get it working and connectable. This is a step that needs to be done before installing. or else steamcmd will install enshrouded in `/root/.local/share/Steam/steamcmd`

To remove it,  use command: `rm -r /root/.local/share/Steam/steamcmd/enshrouded/`

Thank you all for likes and emotes :)

**Comment by DeathGhosts** [comment]
Added streamcmd commandline update suggestion in guide for unpriv containers:
```
#!/bin/sh
/usr/games/steamcmd +@sSteamCmdForcePlatformType windows +force_install_dir /home/steam/enshrouded +login anonymous +app_update 2278520 +quit
```

**Comment by Johnt5818** [comment]
### Some issues that I ran into when I used this guide and what I did to fix them. 

> "Note: if you had the same problem, don't just copy past all of what I did as it will probably not work for you. read it and then make the updated as needed."

When trying to launch my server with `cd //home/steam/enshrouded && ./StartEnshroudedServer.sh` 
I got a: "wine64: command not found" error
I was able to launch it using  `cd //home/steam/enshrouded && wine /home/steam/enshrouded/enshrouded_server.exe`
I later realized that this probably messed up some file permissions that I had to fix later as it launched the server as the root server (I go into that below).

to fix how the game was launched after a reboot I updated the StartEnshroudedServer.sh
`nano //home/steam/enshrouded/StartEnshroudedServer.sh`
on the last line update "wine64" to "wine". should look like this.
```
#!/bin/sh
export WINEARCH=win64
#export WINEPREFIX=/home/steam/.enshrouded_prefix
#export WINEDEBUG=-all
wine /home/steam/enshrouded/enshrouded_server.exe
```

I restarted and checked systemctl `systemctl status enshrouded.service` to see the status of the enshrouded server and found it errored out.
checking the journal `journalctl -t StartEnshroudedServer.sh` I found that there were permission issues with opening the Enshrouded Log files. This was the result of launching the server as the root user at the beginning.
so, I issued: `chown -R steam:steam /home/steam/enshrouded` and that fixed that. I restarted the systemctl `systemctl restart enshrouded.service` and it came up properly this time.

Finally, I also found that it did not take the password that I had entered (don't know why). so I edited the config file that can be found here: `//home/steam/enshrouded/enshrouded_server.json`
While I was in there, I updated a few other settings as I was already intending to do so. here is the info on what the different settings do if you are interested: [Enshrouded Server Gameplay Settings](https://enshrouded.zendesk.com/hc/en-us/articles/20453241249821-Server-Gameplay-Settings)

I hope this helps someone else.  
@DeathGhosts thanks for this. It helped me get this up and running.
